one more (without ebidences but with disemvowel and ooc-mute)

shogunator1337 opened the pull request:
Reworked evidences, added iniswap-blocking, reworked targeting-system, ips and hdids united under unique 'ipid' which improving an anonimity without worsening of moder's privileges (all useless ip and hdid stuff was removed same as rp-mode)

shogunator1337 committed:
a little buggy update

reworked evidences, targeting, hdid-stuff. Reworked client.id and now thats an unique key for targeting.

shogunator1337 committed:
clear config

also in last update was added "ipid" as a mask for ips. Thats easier to input and it's providing a phantom anonimity

shogunator1337 committed:
a little bit more clearing

subj

shogunator1337 committed:
iniswaps

added configurable iniswap-blocking, edited /area

shogunator1337 committed:
a little fixing

fixed inviting issues

shogunator1337 committed:
Merge branch 'master' of https://github.com/Mariomagistr/tsuserver3

shogunator1337 committed:
a little update again

also removed rpmode bcs thats useless.

shogunator1337 committed:
evidence :^)

...

shogunator1337 committed:
clearing

removed pycache

shogunator1337 committed:
MORE CLEARING

here no "ebidence" for sure

shogunator1337 committed:
some changes

added /undj, /unundj, music flood-guard.

shogunator1337 committed:
update README

subj

shogunator1337 committed:
grammar issue

shogunator1337 committed:
fixxx

shogunator1337 committed:
iniswap fix

changed iniswap-allowed to iniswap_allowed

shogunator1337 committed:
More evidence mods

'mods' and 'HiddenCM' is finished.

shogunator1337 committed:
upd

optimization, fixed bug with changing evidence_mod

shogunator1337 committed:
enum stuff

just good practice...

shogunator1337 committed:
:^)

wtf i did before...

shogunator1337 committed:
cleaning

cleaned print() stuff

shogunator1337 committed:
clean up

cleaning pos after character changing

oldmud0 commented on server/masterserverclient.py:
```diff
@@ -42,23 +42,38 @@ def __init__(self, server):
+                        await self.send_raw_message('PING#%')
+                    if cmd == 'PONG':
+                        fl = False
+                    elif cmd == b'NOSERV':
```
wait so why is this one prefixed with `b` and not the other ones

oldmud0 commented on server/client_manager.py:
```diff
@@ -67,12 +86,36 @@ def change_character(self, char_id, force=False):
                 raise ClientError('Invalid Character ID.')
             if not force and not self.area.is_char_available(char_id):
                 raise ClientError('Character not available.')
+            elif not self.area.is_char_available(char_id):
```
probably might want to wind this if block differently:
```python
if not self.area.is_char_available(char_id):
  if force:
    # Kick everyone else off that character
    # ...
  else:
    raise ClientError('Character not available.')

shogunator1337 replied:
fairly

oldmud0 commented on config_sample/areas.yaml:
```diff
@@ -1,9 +1,26 @@
   bglock: true
+  evidence_mod: FFA
+  locking_allowed: false
+  iniswap_allowed: true
```
will old configs work, even without these new settings?

shogunator1337 replied:
sure

oldmud0 reviewed (approved):
this code base has gotten pretty ugly. but you should start testing this to see if it works before I merge

shogunator1337 commented:
What do you mean by 'ugly'? I believe that it's more beatiful than before. But i should add some get_evi_id() (instead of evi_list[client.evi_list[..]]) stuff for sure. 8 people (average) tested this version, it works fine (we tested it for 2 days, i fixed everything we found (thats why here so much commits actually)).

HiddenCM works like evidences on serverD before but with realtime-editing (present is presenting evidence for everyone).

shogunator1337 committed:
Finishing touches

fixed b'noserver', restructured change_character, logging of missed iniswaps.yaml

shogunator1337 commented:
here it is.

shogunator1337 committed:
evidence_mod added

now all 4 evidence_mods works like they supposed to do.

shogunator1337 committed:
back-compatibility

compatibility with old config.yaml

shogunator1337 commented:
Tested all targetings, HiddenCM works fine (here a problem that if u didn't ever wrote something after logging then ur pos is '' and u didn't see any evidence but here 0 chances to fix it except making default pos for every char), Mods works fine (by now), FFA and CM worked before. Iniswap-blocking worked laggy but i fixed it yesterday so it works now. Music flood-guard works ok, disabled if config is outdated. Locks for rooms and inviting-system works ok. Here were no way for player to break it, but now it's hard to break for server-owner too :^)

This build is started on FOC so here a live-test anyways 🤔 

If here will be 0 commits for 24 hours then it's stable build i believe.

oldmud0 commented:
ugly, as in, the combined changes of everyone have caused a necessity for a little bit of refactoring

shogunator1337 committed:
grammar issues

edited undj to blockdj, edited 'evidences' in readme (i don't sure that here a reason for edits in code)

oldmud0 commented on server/aoprotocol.py:
```diff
@@ -414,7 +414,7 @@ def net_cmd_mc(self, args):
                 return
             if not self.client.is_dj:
-                self.client.send_host_message('You are was undj\'d by a moderator.')
+                self.client.send_host_message('You are was blockdj\'d by a moderator.')
```
`You were`*

oldmud0 commented on README.md:
```diff
@@ -108,12 +110,20 @@ Install PyYAML by doing
+        * **CM**
+            - Only CM (case-maker, look at /cm for more info) or moderators can add, edit or remove evidence.
+        * **HiddenCM**
+            - Same as CM, but every evidence have his "owner's position" that can be changed by CM or moderator, what means that only one side can see it (except for 'all'). After presenting evidence once his position changing to 'all'. Possible positions: 'def', 'pro', 'wit', 'jud', 'pos' (means that no one can see this evidence), 'all' (means that everyone can see this evidence).
```
Not a very grammatically correct explanation. Try this:

> **HiddenCM**: Same as CM, but every evidence has a preset "owner's position" which can be set by a CM or moderator, such that only one side/position of the court may see the evidence. After presenting the evidence, the position of the evidence changes to "all." Possible positions include `def` (defense), `pro` (prosecutor), `wit` (witness), `jud` (judge), `pos` (hidden from everyone), and `all` (everyone can see the evidence).

shogunator1337 replied:
thx, fixed

shogunator1337 committed:
grammar issues

thx to longbyte :*
